applicationset/controllers/applicationset_controller_test.go

```diff
@@ -41,6 +41,39 @@ import (
 	"github.com/argoproj/argo-cd/v2/util/settings"
 )
 
+// getDefaultTestClientSet creates a Clientset with the default argo objects
+// To add new objects, use kubeclientset.Tracker().Add(obj)
+func getDefaultTestClientSet() *kubefake.Clientset {
+	argoCDSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      argocommon.ArgoCDSecretName,
+			Namespace: "argocd",
+			Labels: map[string]string{
+				"app.kubernetes.io/part-of": "argocd",
+			},
+		},
+		Data: map[string][]byte{
+			"admin.password":   nil,
+			"server.secretkey": nil,
+		},
+	}
+
+	emptyArgoCDConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      argocommon.ArgoCDConfigMapName,
+			Namespace: "argocd",
+			Labels: map[string]string{
+				"app.kubernetes.io/part-of": "argocd",
+			},
+		},
+		Data: map[string]string{},
+	}
+
+	objects := append([]runtime.Object{}, emptyArgoCDConfigMap, argoCDSecret)
+	kubeclientset := kubefake.NewSimpleClientset(objects...)
```

Comment:
NewSimpleClientset is deprecated, you can use NewClientset. Should work fine as a drop-in replacement.